colpali_engine/collators/custom_collator.py

```diff
@@ -124,9 +140,22 @@ def forward_vision_idefics(self, examples):
         elif any([t is None for t in texts_query]):
             raise ValueError("Some queries are None. This collator does not support None queries yet.")
         else:
-            batch_query = self.processor(
-                text=texts_query, return_tensors="pt", padding="longest", max_length=self.max_length
+            if self.processor.__class__.__name__ == "Idefics3Processor":
+                batch_query = self.processor(
+                images=images,  # NOTE: the image is not used in batch_query but it is required for calling the processor
+                text=texts_query,
+                return_tensors="pt",
+                padding="longest",
+                max_length=self.max_length + self.processor.image_seq_len,
             )
+                del batch_query["pixel_values"]
+                batch_query["input_ids"] = torch.cat((batch_query["input_ids"][..., :batch_query["input_ids"].shape[1] - self.processor.image_seq_len -7], batch_query["input_ids"][..., -1:]), dim=1)
```

Comment:
very hard to undertand - any possibility to comment this so we get what's going on ?
decompose it further ?